### PR TITLE
LPS-134590 - Create a new theme based off of Classic with a basic structure for solutions.

### PR DESCRIPTION
**Pull request by ethib137**
Hey Eduardo, 

It has been a long time since we initially discussed for building a theme for solutions, but finally the pieces have come into place to move forward with this. 

This PR contains the new solutions theme with no custom styles yet added. It is simply built using the Yeoman theme generator, with additional changes as seen in the commits.

@jbalsas it may be helpful to have you verify that this aligns with the way you were describing for us to build a theme based off of classic. 

Thanks guys, I appreciate your help.

PS. It's been quite a while since I've been committing to master, so I'm sure there will be plenty of best practices I'm no longer aware of. Please do provide me lots of feedback and comments for me to make changes as necessary so that I can get up to speed as quickly as possible.

### CODE REVIEW

**Comment by ethib137** [comment]
ci:test

**Comment by epgarcia** [comment]
@ethib137 thanks for your contribution. I'll start reviewing this PR with @marcoscv-work and provide our feedback.

**Comment by marcoscv-work** [comment]
Hey guys!

About the teme, it looks good for me since you guys are just creating the base theme.

As a suggestion, we could add your contributions as much as possible to the Classic theme, I mean, if we have some tokens and some functionalities which look good and make sense I think it will be easy to contribute to the product and ideally use only classic (or the minimal contributions on top of classic).

**Comment by ethib137** [comment]
Thanks @marcoscv-work 

I would like to contribute to classic as much as possible, but I'm unsure how much will be possible without changes to lexicon or the solution designs. For instance, the first component I plan on adding new styling for is buttons. In solutions primary and secondary only differ in color, but in lexicon primary is solid and secondary is an outline. This makes it much more difficult to provide configurable styles for buttons. When solutions needs a button that has an outline this is achieved through the Ghost button type. You can see the button types [here](https://www.figma.com/file/Ekz3GUXzcTSKrsDTYmmBs8/Dialect?node-id=958%3A0). This means that we need very different tokens to style these buttons then would be needed with the default lexicon buttons. We only need tokens for the colors listed under "Action Primary", "Action Secondary", and "Action Neutral" [here](https://www.figma.com/file/Ekz3GUXzcTSKrsDTYmmBs8/Dialect?node-id=1025%3A8013). 

For now this means we need different tokens, but hopefully there are plans for the solution designers and lexicon designers to try to find ways to make their stuff work together better.

Additionally, styling buttons solutions requires overriding lots of the button sass variables. Are there any plans to provide a way to do this without writing a theme? In my next pull I'll be overriding the following variables to style buttons:
```
$btn-border-radius-sm: 4px;
$btn-border-radius: 6px;
$btn-border-radius-lg: 8px;

$btn-font-size-sm: 0.875rem; // 14px
$btn-font-size: 1rem; // 16px
$btn-font-size-lg: 1.125rem; // 18px

$btn-line-height-sm: calc(20/14); // 20px
$btn-line-height: calc(24/16); // 24px
$btn-line-height-lg: calc(28/18); // 28px

$btn-padding-y-sm: 0.5rem; // 8px
$btn-padding-y: 0.75rem; // 12px
$btn-padding-y-lg: 0.875rem; // 14px

$btn-padding-x-sm: 0.75rem; // 12px
$btn-padding-x: 1rem; // 16px
$btn-padding-x-lg: 1.125rem; // 18px

$inline-item-spacer-x: 0.25em; // 4px
```

Let me know if you think it would be helpful to have a call to discuss this more.

**Comment by marcoscv-work** [comment]
Thanks @ethib137 

Now I see it more clear!

So finally the objective of the theme is to implement Dialect, right?
For this, yeah, we would need to implement new buttons and probably a lot of new things that probably do not need to be based on Clay, so I agree and we go with a new theme to content all new things.
And of course, If some of what we implement in this theme is interesting to be moved to Classic we can do it

**Comment by ethib137** [comment]
@marcoscv-work yep, that's all correct. 

This theme will implement the designs from Dialect. The goal is that this should primarily be done by styling Clay, but just like another product team, when solution's need a component that doesn't exist in Clay/Lexicon it will be implemented first for solutions only, and then possibly brought back to Clay/Lexicon if it would be helpful to be shared.

File: modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/_mixins.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	} @else if $type == "fluffy-5" {
+		box-shadow: 0px 15px 60px $color;
+	} @else if $type == "fluffy-6" {
+		box-shadow: 0px 17.5px 70px $color;
+	} @else if $type == "fluffy-7" {
+		box-shadow: 0px 35px 90px $color;
+	} @else {
+		@error "Unknown elevation: #{$type}.";
+	}
+}
```
Comment (marcoscv-work) [inline modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/_mixins.scss]: I suppose this part match with Dialect type of shadows but wouldn't be better if we use a custom property for shadow with the format: `Xpx Xpx Xpx` and another one for shadow color?
Then we can combine them, example:

```css
--your-box-shadow-color: XX;
--your-box-shadow: XX;

box-shadow: var(--your-box-shadow) var(--your-box-shadow-color);
```

Just thinking about a maintenance and minimal code possible but the mixin can work great anyway

File: modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/components/_buttons.scss [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+	&:not(.btn-sm):not(.btn-lg) {
+		height: 48px;
+		width: 48px;
+	}
+
+	&.btn-lg {
+		height: 56px;
+		width: 56px;
+	}
+}
```
Comment (marcoscv-work) [inline modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/components/_buttons.scss]: Probably you could use SASS vaiables:

https://github.com/liferay/clay/blob/master/packages/clay-css/src/scss/atlas/variables/_buttons.scss#L32-L42

File: modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/components/_buttons.scss [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+	opacity: 1;
+}
+
+.btn {
+	border-width: 0;
+}
+
+// Solid
+
+:not(.btn-ghost):not(.btn-borderless) {
```
Comment (marcoscv-work) [inline modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/components/_buttons.scss]: Just adding some ideas/possibilities when we need to add complex variables combinations for buttons:

https://github.com/edalgrin/clay/pull/3/files

In this example, we create a generic .btn class, and then we allow to overwrite them by creating a new class containing custom properties, for example:

```css
.btn-ghost {
	--btn-border-radius: X;
	--btn-border-width: X;
	--btn-color: X;
	--btn-font-size: X;
	--btn-padding-y: X;
	--btn-padding-x: X;
}
```

of course, they could be based on a new token:

`--btn-padding-x: var(--my-theme-token);`

**Comment by epgarcia** [comment]
Notwithstanding the comments by @marcoscv-work, we'll forward the PR to Brian. I opened a new PR to run the tests, before we execute the forward. See #1035

File: modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/components/_buttons.scss (hunk 2 above)
Comment (ethib137) [reply modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/components/_buttons.scss]: Perfect! I wasn't aware of those variables. Thanks. :)

File: modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/_mixins.scss (hunk 1 above)
Comment (ethib137) [reply modules/apps/frontend-theme/frontend-theme-solutions/src/css/solutions/_mixins.scss]: Thanks Marco, this results in simpler code. I like it.

**Pull request closed without merging**